dev/system-test/firestore.ts

```diff
@@ -711,13 +716,20 @@ describe('DocumentReference class', () => {
       });
   });
 
-  it('enforces that updated document exists', () => {
-    return randomCol
-      .doc()
-      .update({foo: 'b'})
-      .catch(err => {
-        expect(err.message).to.match(/No document to update/);
-      });
+  it('enforces that updated document exists', async () => {
+    await expect(
+      randomCol
+        .doc()
+        .update({foo: 'b'})
+        .catch(err => {
+          // The emulator generates a different error message, do not validate.
+          if (process.env.FIRESTORE_EMULATOR_HOST === undefined) {
+            expect(err.message).to.match(/No document to update/);
+          }
+
+          throw err;
```

Comment:
Is this test going to unconditionally pass, since the `catch` block always throws and the test is verifying that the promise is rejected?

Perhaps a more robust way to write this test would be to save the promise to a local variable. Then, if running against the Firestore emulator just verify that the promise is rejected without validating its error message. Otherwise, if running against production, verify that it is rejected with the specified message. This is basically what is being done on lines 2498-2506.

This comment also applies to the test `will fail to delete document with exists: true if doc does not exist` below.

Reply:
Nice catch. I wasn't thinking through how chai assertions work. Both instances have been updated. PTAL.